lib/internal/bootstrap_node.js

```diff
@@ -167,7 +167,10 @@
 
         const internalModule = NativeModule.require('internal/module');
         internalModule.addBuiltinLibsToObject(global);
-        evalScript('[eval]');
+        if (process.binding('config').mainMode === 'esm')
+          evalModule();
+        else
+          evalScript('[eval]');
```

Comment:
we could potentially put a warning here if we want to eventually change the default loader.